src/Board.js

```diff
@@ -1,56 +1,57 @@
 import { useState } from 'react'
 import Square from './Square.js'
 
-const board = [...Array(10).keys()]
-const filled = 'red'
-const unfilled = 'grey'
+const board = [...Array(34).keys()]
+const filled = 'black'
+const unfilled = 'white'
 
-export default function Board(props) {
-  const [squareColors, setSquareColors] = useState(Array(board.length * board.length).fill(unfilled))
-  const [startSquare, setStartSquare] = useState(null)
-  const [endSquare, setEndSquare] = useState(null)
-  const [drawColor, setDrawColor] = useState(filled)
+export default function Board() {
+  const [allState, setAllState] = useState({
+    colors: Array(board.length * board.length).fill(unfilled),
+    startSquare: null,
+    endSquare: null,
+    drawColor: filled
+  })
 
-  const handleSquareClick = (i) => {
-    const color = (squareColors[i] === unfilled) ? filled : unfilled
-    setDrawColor(color)
-    fillSquare(i, color)
-    setStartSquare(i)
-    setEndSquare(i)
-  }
+  const handleSquareClick = (i) => 
+    setAllState(prev => {
```

Comment:
OOOH nice! I didn't realize the previous state was accessible like this.

Reply:
It's way nice. It was the only way I could get memoization to work, since I only pass this function to the Squares once and never update it. If I just use `allState` in here instead, it captures its original value and is never updated. That was fun to track down.